packages/webpack/src/plugin.js

```diff
@@ -464,50 +464,72 @@ class LavaMoatPlugin {
                   )
                   return
                 }
-                diag.rawDebug(
-                  1,
-                  '> adding runtime (additionalChunkRuntimeRequirements)'
+                let runtimeChunks = []
+                if (
+                  chunk.name &&
+                  options.unlockedChunksUnsafe?.test(chunk.name)
+                ) {
+                  diag.rawDebug(
+                    1,
+                    `> adding UNLOCKED runtime for chunk ${chunk.name}`
+                  )
+                  runtimeChunks = [
+                    {
+                      name: 'ENUM',
+                      file: path.join(__dirname, './ENUM.json'),
+                      json: true,
+                    },
+                    {
+                      name: 'runtime',
+                      file: path.join(
+                        __dirname,
+                        './runtime/runtimeUnlocked.js'
+                      ),
+                    },
+                  ]
+                } else {
+                  diag.rawDebug(1, `> adding runtime for chunk ${chunk.name}`)
+                  // narrow down the policy and map to module identifiers
+                  const policyData = identifierLookup.getTranslatedPolicy()
+
+                  runtimeChunks = [
+                    {
+                      name: 'root',
+                      data: identifierLookup.root || null,
+                      json: true,
+                    },
+                    {
+                      name: 'idmap',
+                      data: identifierLookup.identifiersForModuleIds || null,
+                      json: true,
+                    },
+                    {
+                      name: 'unenforceable',
+                      data: identifierLookup.unenforceableModuleIds || null,
+                      json: true,
+                    },
+                    { name: 'options', data: runtimeOptions, json: true },
+                    { name: 'policy', data: policyData, json: true },
+                    {
+                      name: 'ENUM',
+                      file: path.join(__dirname, './ENUM.json'),
+                      json: true,
+                    },
+                    {
+                      name: 'endowmentsToolkit',
+                      shimRequire: 'lavamoat-core/src/endowmentsToolkit.js',
```

Comment:
I'm guessing this is found via Node.js module resolution?  if so, it may be better to "fail fast" and `require.resolve` it.

Reply:
shimRequire is passed straight into require.resolve in `assembleRuntime`